File: backend/libs/entities/src/main/java/bencoepp/livius/entities/info/Changelog.java
package bencoepp.livius.entities.info;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.Transient;
import org.springframework.data.mongodb.core.mapping.Document;
import javax.xml.crypto.dsig.keyinfo.KeyValue;
import java.util.ArrayList;
import java.util.List;

/**
 * The Changelog class represents a changelog document.
 * It is used to store information about software updates and changes.
 */
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Document(collection = "changelogs")
public class Changelog {
    @Transient
    public static final String SEQUENCE_NAME="changelog_seq";
    @Id
    private String id;
    /**
     * The version variable represents the version number of a software update or change.
     * It is used in the Changelog class to track the version of each update or change.
     * This variable is a String type.
     */
    private String version;
    /**
     * The title variable represents the title of a changelog entry.
     * It is used in the Changelog class to store the title of each update or change.
     * This variable is a String type and is marked as private.
     */
    private String title;
    private String description;
    private List<String> changes;
    /**
     * The contributors variable represents a list of contributors to a changelog entry.
     * It is used in the Changelog class to store the names of individuals who have contributed to the update or change.
     * This variable is of type List<String> and is marked as private, indicating that it can only be accessed within the Changelog class.
     */
    private List<String> contributors = new ArrayList<>();
    /**
     * The acknowledgement variable represents the acknowledgment given for a changelog entry.
     * It is used in the Changelog class to store the acknowledgment related to each update or change.
     * This variable is a String type and is marked as private.
     */
    private String acknowledgement;
    private String references;
    private String created;
    /**
     * The updated variable represents the date and time when the changelog entry was last updated.
     * It is used in the Changelog class to store the updated timestamp of each entry.
     * This variable is of type Instant, which is a representation of a moment in time.
     * It is marked as private, indicating that it can only be accessed within the Changelog class.
     */
    private String updated;
}

File: frontend/lib/model/info/changelog.dart
class Changelog {
  late String id;
  late String version;
  late String title;
  late String description;
  late List<String> changes;
  late List<String> contributors;
  late String acknowledgement;
  late String references;
  late String created;
  late String updated;

  Changelog.fromJson(Map<String, dynamic> json)
      : id = json['id'] as String,
        version = json['version'] as String,
        title = json['title'] as String,
        description = json['description'] as String,
        changes =
            (json['changes'] as List).map((item) => item as String).toList(),
        contributors = (json['contributors'] as List)
            .map((item) => item as String)
            .toList(),
        acknowledgement = json['acknowledgement'] as String,
        references = json['references'] as String,
        created = json['created'] as String,
        updated = json['updated'] as String;

  Map<String, dynamic> toJson() => {
        'id': id,
        'version': version,
        'title': title,
        'description': description,
        'changes': changes,
        'contributors': contributors,
        'acknowledgement': acknowledgement,
        'references': references,
        'created': created,
        'updated': updated,
      };
}
